packages/vite/src/node/utils.ts

```diff
@@ -108,7 +108,7 @@ export const isDataUrl = (url: string): boolean => dataUrlRE.test(url)
 
 const knownJsSrcRE = /\.((j|t)sx?|mjs|vue|marko|svelte)($|\?)/
 export const isJSRequest = (url: string): boolean => {
-  if (knownJsSrcRE.test(url)) {
+  if (!url.includes('?') && knownJsSrcRE.test(url)) {
```

Comment:
I think moving the `url = cleanUrl(url)` up would have the same effect

Reply:
I don't think so.
If only moving the url = cleanUrl(url)，the first if condition "knownJsSrcRE.test(url)" is still truthy, and isJSRequest still returns true.

Reply:
I mean move this code ahead of the first condition, it will remove the querystring of the url.

Reply:
You are right, that would be better

Reply:
code updated